lib/phobos.rb

```diff
@@ -31,19 +31,46 @@
 
 Thread.abort_on_exception = true
 
+Logging.init :debug, :info, :warn, :error, :fatal
+
+# Monkey patch to fix this issue: https://github.com/zendesk/ruby-kafka/pull/732
+module Logging
+  # :nodoc:
+  class Logger
+    # :nodoc:
+    def formatter=(*args); end
+
+    # :nodoc:
+    def push_tags(*args); end
+
+    # :nodoc:
+    def pop_tags(*args); end
+  end
+end
+
 module Phobos
   class << self
     attr_reader :config, :logger
     attr_accessor :silence_log
 
     def configure(configuration)
       @config = fetch_configuration(configuration)
-      @config.class.send(:define_method, :producer_hash) { Phobos.config.producer&.to_hash }
-      @config.class.send(:define_method, :consumer_hash) { Phobos.config.consumer&.to_hash }
+      config_define_methods
       @config.listeners ||= []
       configure_logger
     end
 
+    def config_define_methods # rubocop:disable Metrics/AbcSize
+      @config.class.send(:define_method, :cache_sync_producer) do
```

Comment:
Would it be enough to call Phobos.config.producer&.cache_sync_producer where we need to read this, and not making this a root level config method?

Reply:
Unfortunately Phobos passes the entire config off to RubyKafka, which fails because it's not defined on the RubyKafka `producer` method. This was my attempt at doing it - there are at least two other ways (separating the configs out, or removing it from the hash at the time we create a new producer instead of here).

Reply:
I'd say go for the latter approach, it follows the existing pattern:

```ruby
      class PublicAPI
        NAMESPACE = :phobos_producer_store
        ASYNC_PRODUCER_PARAMS = [:max_queue_size, :delivery_threshold, :delivery_interval].freeze
        INTERNAL_PRODUCER_PARAMS = [:cache_sync_producer].freeze

        # ...

        def regular_configs
          Phobos.config.producer_hash
            .reject { |k, _| ASYNC_PRODUCER_PARAMS.include?(k) }
            .reject { |k, _| INTERNAL_PRODUCER_PARAMS.include?(k) }
        end

        def async_configs
          Phobos.config.producer_hash
            .reject { |k, _| INTERNAL_PRODUCER_PARAMS.include?(k) }
        end
```